winnery_ticket_experiment.py

```diff
@@ -0,0 +1,86 @@
+from keras.callbacks import Callback
+from kardashigans.simple_experiment import SimpleExperiment
+from kardashigans.trainer import BaseTrainer
+
+
+class ZeroWeightsCallback(Callback):
+    """A Callback that sets the given values to 0.0 on every epoch.
+    This is equivalent to removing the given weights.
+    """
+
+    def __init__(self, zero_weight_mask):
+        super(ZeroWeightsCallback, self).__init__()
+        self.zero_weight_mask = zero_weight_mask
+
+    def _zero_masked_weights(self):
+        for i in range(1, len(self.model.layers)):
```

Comment:
I think you should provide the list of weighted layer to the callback on construction. The trainer class constructs the list, use it when creating the callback?

Reply:
Done.